pages/availability/troubleshoot.tsx

```diff
@@ -30,23 +26,29 @@ export default function Troubleshoot({ user }) {
   }
 
   const fetchAvailability = (date) => {
-    fetch(
-      `/api/availability/${session.user.username}?dateFrom=${date
-        .startOf("day")
-        .utc()
-        .startOf("day")
-        .format()}&dateTo=${date.endOf("day").utc().endOf("day").format()}`
-    )
+    const dateFrom = date.startOf("day").utc().format();
+    const dateTo = date.endOf("day").utc().format();
+
+    fetch(`/api/availability/${user.username}?dateFrom=${dateFrom}&dateTo=${dateTo}`)
       .then((res) => {
         return res.json();
       })
-      .then((apires) => setAvailability(apires))
+      .then((availableIntervals) => {
+        setAvailability(availableIntervals);
+        setLoading(false);
+      })
       .catch((e) => {
         console.error(e);
       });
   };
 
-  fetchAvailability(selectedDate);
+  useEffect(() => {
+    fetchAvailability(selectedDate);
+  });
```

Comment:
Regarding your fix for the API-flooding loop:
I tested your code (current state from your branch) and it doesn't fix it, at least for me - it still refreshes forever and floods the API. However, when watching `selectedDate` in the useEffect closure, it works perfectly for me. :)

```suggestion
  }, [selectedDate]);
```